Automattic/wp-calypso
# Magic Login: Extract components from general login & update to match changes there

This change moves the Magic Login content to its own component (instead of commingling with the `WPLogin` component). The idea is to reduce the size of the latter and better separate concerns.

Continuation of the work started in #14027.

This is a preliminary change to make 2fa integration easier across the board.
This is currently behind a development feature flag & still has some rough edges which will be polished in follow-up PRs.

## To Test:
* `npm run test-client client/state` -- all state tests should still pass
* Visit http://calypso.localhost:3000/log-in in an incognito window -- you should see an "Email me a login link" link
* Clicking the above link should take you to http://calypso.localhost:3000/log-in/link & you should see a prompt for your email address (with some known styling issues which will be addressed in a follow up PR)
